objectRecognitionSource/src/main/java/edu/ml/tensorflow/util/ImageUtil.java

```diff
@@ -101,32 +89,81 @@ public void saveImage(final BufferedImage image, final String target) {
-                                    super.onSuccess(body, response);
-                                    LOGGER.trace("Content Location = " + this.getBodyAsJsonObject().get("content"));
-                                    if(imgFile.delete()) { 
-                                        LOGGER.trace("Temp file deleted successfully"); 
-                                    } else { 
-                                        LOGGER.warn("Failed to delete temp file"); 
-                                    } 
-                                }
-                                
-                                @Override public void onError(List<VantiqError> errors, Response response) {
-                                    super.onError(errors, response);
-                                    LOGGER.error("Errors uploading image with VANTIQ SDK: " + errors);
-                                }
-                    });
+                    uploadImage(imgFile, target);
                 } catch (IOException e) {
                     LOGGER.error("Unable to save image {}", target, e);
                 }
             }
         }
     }
     
+    /**
+     * A method used to upload images to VANTIQ, using the VANTIQ SDK
+     * @param imgFile   The file to be uploaded. If not specified to save locally, this file will be deleted.
+     * @param target    The name of the file to be uploaded.
+     */
+    public void uploadImage(File imgFile, String target) {
+        if (queryResize) {
+            try {
+                BufferedImage resizedImage = resizeImage(ImageIO.read(imgFile));
+                File tmpFile = File.createTempFile("tmp", ".jpg");
+                tmpFile.deleteOnExit();
+                ImageIO.write(resizedImage, "jpg", tmpFile);
+                vantiq.upload(tmpFile, 
+                        "image/jpeg", 
+                        "objectRecognition/" + sourceName + '/'  + target,
+                        new BaseResponseHandler() {
+                            @Override public void onSuccess(Object body, Response response) {
+                                super.onSuccess(body, response);
+                                LOGGER.trace("Content Location = " + this.getBodyAsJsonObject().get("content"));
+                                
+                                if (outputDir == null) {
+                                    deleteImage(imgFile);
+                                }
+                            }
+                            
+                            @Override public void onError(List<VantiqError> errors, Response response) {
+                                super.onError(errors, response);
+                                LOGGER.error("Errors uploading image with VANTIQ SDK: " + errors);
+                            }
+                });
+            } catch (IOException e) {
+                LOGGER.error("An error occured while reading the locally saved image file. " + e.getMessage());
+            }
```

Comment:
Seems like most of this code is duplicated.  Couldn't this do
```
fileToUpload = imgFile
if (queryResize) {
    ... do the resize...
    fileToUpload = tmpFile
} 
// Now do the upload of fileToUpload
```

Reply:
The reason I implemented this with redundant code is because I was getting the following error on line 121 (deleteImage(imgFile) - "Local variable statement defined in an enclosing scope must be final or effectively final."

I can't re-assign the value of imgFile because then the onSuccess callback function cannot delete the temporary file if it needs to. Let me know if that makes sense!

Reply:
Yeah, don't try & reassign the parameter -- confusing anyway, if call by value vs name vs reference, etc.

IntelliJ seems happy with
```
   public void uploadImage(File imgFile, String target) {
        File fileToUpload = imgFile;
        if (queryResize) {
            try {
                BufferedImage resizedImage = resizeImage(ImageIO.read(imgFile));
                File tmpFile = File.createTempFile("tmp", ".jpg");
                tmpFile.deleteOnExit();
                ImageIO.write(resizedImage, "jpg", tmpFile);
                fileToUpload = tmpFile;

            } catch (IOException e) {
                LOGGER.error("An error occured while reading the locally saved image file. " + e.getMessage());
            }
        } 
        vantiq.upload(fileToUpload,
                "image/jpeg",
                "objectRecognition/" + sourceName + '/'  + target,
                new BaseResponseHandler() {
                    @Override public void onSuccess(Object body, Response response) {
                        super.onSuccess(body, response);
                        LOGGER.trace("Content Location = " + this.getBodyAsJsonObject().get("content"));

                        if (outputDir == null) {
                            deleteImage(imgFile);
                        }
                    }

                    @Override public void onError(List<VantiqError> errors, Response response) {
                        super.onError(errors, response);
                        LOGGER.error("Errors uploading image with VANTIQ SDK: " + errors);
                    }
                });
    }
```